src/reducers/documentReducer.js

```diff
@@ -0,0 +1,22 @@
+import { UPDATE_MARKDOWN, ADD_TAB, DELETE_TAB } from '../actions/documentActions';
+
+const initialState = {
+  active: 'default',
+  markdowns: {
+    default: '# hi there',
+  }
+};
+
+export default function reducer(state = initialState, action) {
+  switch(action.type) {
+    case UPDATE_MARKDOWN:
+      return { ...state, markdowns: { ...state.markdowns, [state.active]: action.payload } };
+    case ADD_TAB:
+      return { ...state, markdowns: { ...state.markdowns, [action.payload]: `# ${action.payload}` } };
+    case DELETE_TAB:
+      delete state.markdowns[action.payload];
+      return { ...state, markdowns: { ...state.markdowns } };
```

Comment:
im curious if this would work if you only spread in state.
return { ...state };

Reply:
It still works when using just { ...state };!